lib/msal-common/src/client/DeviceCodeClient.ts

```diff
@@ -147,62 +147,69 @@ export class DeviceCodeClient extends BaseClient {
         const deviceCodeExpirationTime = TimeUtils.nowSeconds() + deviceCodeResponse.expiresIn;
         const pollingIntervalMilli = deviceCodeResponse.interval * 1000;
 
+        /**
+         * Breaks the polling with specific conditions.
+         * @param request CommonDeviceCodeRequest
+         * @param deviceCodeResponse DeviceCodeResponse
+         */
+        const continuePolling = (): boolean => {
+            if (request.cancel) {
+                this.logger.error("Token request cancelled by setting DeviceCodeRequest.cancel = true");
+                throw ClientAuthError.createDeviceCodeCancelledError();
```

Comment:
Do these errors throw and halt the application or are they supposed to bubble up in a promise as an error response to the client request?

Reply:
Never mind. We are doing `await delay()` below which should bubble this up.